src/app/shared/filter-by-property/filter-by-property.pipe.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Pipe, PipeTransform} from '@angular/core';
+
+@Pipe({
+    name: 'filterByProperty',
+})
+export class FilterByPropertyPipe implements PipeTransform {
+    transform<T extends Array<Record<string, any>>, P extends keyof T[0], V extends T[0][P]>(
+        array: T,
+        property: P,
+        value: V,
+    ) {
```

Comment:
Смотри, можно поступить чуть хитрее и обозначить дженериком `T` не массив сущностей, с которыми планируется работу, а саму сущность, с которой пларнируеш работать:
```suggestion
    transform<T extends Record<string, any>, P extends keyof T, V extends T[P]>(
        array: T[],
        property: P,
        value: V,
    ) {
```
Так, мне кажется, получается чуть понятней и прозрачней

Reply:
@Letto228 принял, спасибо!